Scripts/Characters/Player/Player.cs

```diff
@@ -43,7 +43,20 @@ private void HandleMovement(double delta)
         Vector2 direction = Input.GetVector("move_left", "move_right", "move_up", "move_down");
         if (direction != Vector2.Zero)
         {
-            _velocity.X = direction.X * Speed;
+
+            // Normalize only if both horizontal and vertical inputs are non-zero
```

Comment:
So I'm just commenting to give you some experience receiving one, I'll approve this as I get it and it make sense, we'll get it pulled in.

But for some salient advice here, this function is getting too long.  At most we want functions to be about 10 lines long, game code can string out and become spaghetti so quickly.  Tightening up the movement is in the plans after the alpha-stage, so we'll come back and address it then, but if you do happen to come back here and are in this, you'll want to pull the logic blocks out into their own functions.

For help with refactoring, schedule a meeting or text me!